src/main/java/com/minecolonies/coremod/entity/ai/citizen/builder/ConstructionTapeHelper.java

```diff
@@ -0,0 +1,289 @@
+                    newY = checkIfPlaceable(i,y,z1,world);
+                    BlockPos row1 = new BlockPos(i, newY, z1);
+                    world.setBlockState(row1, ModBlocks.blockConstructionTape.getDefaultState().withProperty(FACING, EnumFacing.SOUTH));
+                    newY = checkIfPlaceable(i,y,z3,world);
+                    BlockPos row2 = new BlockPos(i, newY, z3);
+                    world.setBlockState(row2, ModBlocks.blockConstructionTape.getDefaultState().withProperty(FACING, EnumFacing.NORTH));
+                }
+            }
+            else
+            {
+                for (int i = x3+1; i < x1; i++)
+                {
+                    newY = checkIfPlaceable(i,y,z1,world);
+                    BlockPos row1 = new BlockPos(i, newY, z1);
+                    world.setBlockState(row1, ModBlocks.blockConstructionTape.getDefaultState().withProperty(FACING, EnumFacing.SOUTH));
+                    newY = checkIfPlaceable(i,y,z3,world);
+                    BlockPos row2 = new BlockPos(i, newY, z3);
+                    world.setBlockState(row2, ModBlocks.blockConstructionTape.getDefaultState().withProperty(FACING, EnumFacing.NORTH));
+                }
+            }
+            if (z1 < z3)
+            {
+                for (int i = z1+1; i < z3; i++)
+                {
+                    newY = checkIfPlaceable(x1,y,i,world);
+                    BlockPos row3 = new BlockPos(x1, newY, i);
+                    world.setBlockState(row3, ModBlocks.blockConstructionTape.getDefaultState().withProperty(FACING, EnumFacing.EAST));
+                    newY = checkIfPlaceable(x3,y,i,world);
+                    BlockPos row4 = new BlockPos(x3, newY, i);
+                    world.setBlockState(row4, ModBlocks.blockConstructionTape.getDefaultState().withProperty(FACING, EnumFacing.WEST));
+                }
+            }
+            else
+            {
+                for (int i = z3+1; i < z1; i++)
+                {
+                    newY = checkIfPlaceable(x1,y,i,world);
+                    BlockPos row3 = new BlockPos(x1, newY, i);
+                    world.setBlockState(row3, ModBlocks.blockConstructionTape.getDefaultState().withProperty(FACING, EnumFacing.EAST));
+                    newY = checkIfPlaceable(x3,y,i,world);
+                    BlockPos row4 = new BlockPos(x3, newY, i);
+                    world.setBlockState(row4, ModBlocks.blockConstructionTape.getDefaultState().withProperty(FACING, EnumFacing.WEST));
+                }
+            }
+            newY = checkIfPlaceable(x1,y,z1,world);
+            BlockPos corner1 = new BlockPos(x1, newY, z1);
+            newY = checkIfPlaceable(x1,y,z3,world);
+            BlockPos corner2 = new BlockPos(x1, newY, z3);
+            newY = checkIfPlaceable(x3,y,z1,world);
+            BlockPos corner3 = new BlockPos(x3, newY, z1);
+            newY = checkIfPlaceable(x3,y,z3,world);
+            BlockPos corner4 = new BlockPos(x3, newY, z3);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Local variable 'corner4' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)